akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -350,6 +350,8 @@ trait FlowOps[+Out, +Mat] {
 
   private final val _identity = (x: Any) ⇒ x
 
+  def recover[T](pf: PartialFunction[Throwable, T]): Repr[T, Mat] = andThen(Recover(pf.asInstanceOf[PartialFunction[Any, Any]]))
```

Comment:
Ah, yes, +1! Using this stage does not prevent losing messages that arrived *before* onError.